components/choose-cinema-dropdown.tsx

```diff
@@ -17,14 +17,20 @@ import { useSearchCinemas } from '@/hooks/useCinemas'
 import { Cinema } from '@/lib/api/types'
 
 interface ChooseCinemaDropdownProps {
+  selectedCinema?: Cinema | null
   onCinemaSelect: (cinema: Cinema) => void
 }
 
 export default function ChooseCinemaDropdown({
+  selectedCinema: externalSelectedCinema,
   onCinemaSelect,
 }: ChooseCinemaDropdownProps) {
   const [isModalVisible, setIsModalVisible] = useState(false)
-  const [selectedCinema, setSelectedCinema] = useState<Cinema | null>(null)
+  const [internalSelectedCinema, setInternalSelectedCinema] =
+    useState<Cinema | null>(null)
+
+  // Use external selectedCinema if provided, otherwise use internal state
+  const selectedCinema = externalSelectedCinema ?? internalSelectedCinema
```

Comment:
I like the idea of making this a controlled component, so you don't need the internal state anymore, the selected can dependend only on the prop `selectedCinema` that is passed from the parent.